### PR TITLE
Fix map view

### PR DESCRIPTION
**Pull request by tanruiquan**
- Map view is user interact-able
- Add a drop pin
- Edit is being reflected

### CODE REVIEW

File: TourMate/TourMate/Frontend/Plan/Views/Common/SearchViewModel.swift [hunk 1]
```diff
@@ -20,7 +20,7 @@ class SearchViewModel: ObservableObject {
 
     private var cancellableSet: Set<AnyCancellable> = []
 
-    init(locationService: LocationService = RealLocationService()) {
+    init(locationService: LocationService = MockLocationService()) {
```
Comment (keanecjy) [inline TourMate/TourMate/Frontend/Plan/Views/Common/SearchViewModel.swift]: Oh, why are changing to use the mock service now?

File: TourMate/TourMate/Frontend/Plan/Views/PlanViews/Components/TimingView.swift [hunk 2]
```diff
@@ -8,8 +8,8 @@
 import SwiftUI
 
 struct TimingView: View {
-    @Binding var plan: Plan
-    @State var dateFormatter = DateFormatter()
+    let plan: Plan
+    @State private var dateFormatter = DateFormatter()
```
Comment (keanecjy) [inline TourMate/TourMate/Frontend/Plan/Views/PlanViews/Components/TimingView.swift:12]: Actually why is the dateformatter a state variable?

**Review by keanecjy — APPROVED** [review]

File: TourMate/TourMate/Frontend/Plan/Views/Common/SearchViewModel.swift (hunk 1 above)
Comment (tanruiquan) [reply TourMate/TourMate/Frontend/Plan/Views/Common/SearchViewModel.swift]: Oh, i didn't want to make so many api calls when testing

File: TourMate/TourMate/Frontend/Plan/Views/PlanViews/Components/TimingView.swift (hunk 2 above)
Comment (tanruiquan) [reply TourMate/TourMate/Frontend/Plan/Views/PlanViews/Components/TimingView.swift:12]: I wouldn't be able to assign to self otherwise

Comment (keanecjy) [reply TourMate/TourMate/Frontend/Plan/Views/PlanViews/Components/TimingView.swift:12]: I think u should actually assign the date formatter on init(). Apparently I was reading and they said onAppear might appear too late on some cases

File: TourMate/TourMate/Frontend/Plan/Views/Common/SearchViewModel.swift (hunk 1 above)
Comment (keanecjy) [reply TourMate/TourMate/Frontend/Plan/Views/Common/SearchViewModel.swift]: Oh ok, maybe should keep it as mock then haha

**Pull request merged**